drivers/wifi/esp8266/wifi_esp8266.c

```diff
@@ -0,0 +1,1570 @@
+}
+
+/* Echo Handler for commands without related sockets */
+static void on_cmd_atcmdecho_nosock(struct net_buf **buf, u16_t len)
+{
+}
+
+static void on_cmd_esp8266_ready(struct net_buf **buf, u16_t len)
+{
+	k_sem_give(&foo_data.response_sem);
+}
+
+static void esp8266_ip_addr_work(struct k_work *work)
+{
+	int ret;
+
+	ret = send_at_cmd(NULL, "AT+CIPSTA_CUR?", MDM_CMD_TIMEOUT);
+	if (ret < 0) {
+		LOG_ERR("failed to get ip address information\n");
+		return;
+	}
+
+	/* update interface addresses */
+	net_if_ipv4_set_gw(foo_data.iface, &foo_data.gw);
+	net_if_ipv4_set_netmask(foo_data.iface,
+			&foo_data.netmask);
+	net_if_ipv4_addr_add(foo_data.iface, &foo_data.ip, NET_ADDR_DHCP,
+		0);
+}
+
+static void on_cmd_ip_addr_get(struct net_buf **buf, u16_t len)
+{
+		k_delayed_work_submit_to_queue(&esp8266_workq,
+			&foo_data.ip_addr_work, K_SECONDS(2));
+}
+
+static void on_cmd_conn_open(struct net_buf **buf, u16_t len)
+{
+	int id;
+	u8_t msg[2];
+
+	net_buf_linearize(msg, 1, *buf, 0, 1);
+	msg[1] = '\0';
+	id = atoi(msg);
+
+	foo_data.socket_data[id].connected = 1;
+}
+
+static void on_cmd_conn_closed(struct net_buf **buf, u16_t len)
+{
+	int id;
+	u8_t msg[2];
+	struct socket_data *sock = NULL;
+
+	net_buf_linearize(msg, 1, *buf, 0, 1);
+	msg[1] = '\0';
+	id = atoi(msg);
+
+	sock = &foo_data.socket_data[id];
+	sock->connected = 0;
+}
+
+static void on_cmd_sent_bytes(struct net_buf **buf, u16_t len)
+{
+	char temp[32];
+	char *tok;
+
+	net_buf_linearize(temp, 32, *buf, 0, len);
+	temp[len] = '\0';
+
+	tok = strtok(temp, " ");
+	while(tok) {
+		tok = strtok(NULL, " ");
+	}
+}
+
+
+static void on_cmd_wifi_scan_resp(struct net_buf **buf, u16_t len)
+{
+	int i;
+	char temp[32];
+	struct wifi_scan_result result;
+	int delimiters[6];
+	int slen;
+
+	/* ecn, ssid, rssi, mac, channel, freq */
+
+	delimiters[0] = 1;
+	for (i = 1; i < 6; i++) {
+		delimiters[i] = net_buf_find_next_delimiter(*buf, ',',
+					delimiters[i-1] + 1);
+		if (delimiters[i] == -1) {
+			return;
+		}
+		delimiters[i]++;
+	}
+
+	/* ecn */
+	net_buf_linearize(temp, 1, *buf, delimiters[0],
+			  delimiters[1] - delimiters[0]);
+	if (temp[0] != '0') {
+		result.security = WIFI_SECURITY_TYPE_PSK;
+	} else {
+		result.security = WIFI_SECURITY_TYPE_PSK;
+	}
+
+	/* ssid */
+	slen = delimiters[2] - delimiters[1] - 3;
+	net_buf_linearize(result.ssid, 32, *buf, delimiters[1] + 1,
+			  slen);
+	result.ssid_length = slen;
+
+	/* rssi */
+	slen = delimiters[3] - delimiters[2];
+	net_buf_linearize(temp, 32, *buf, delimiters[2], slen);
+	temp[slen] = '\0';
+	result.rssi = strtol(temp, NULL, 10);
+
+	/* channel */
+	slen = delimiters[5] - delimiters[4];
+	net_buf_linearize(temp, 32, *buf, delimiters[4], slen);
+	temp[slen] = '\0';
+	result.channel = strtol(temp, NULL, 10);
+
+	/* issue callback to report scan results */
+	if (foo_data.wifi_scan_cb) {
+		foo_data.wifi_scan_cb(foo_data.iface, 0, &result);
+	}
+}
+
+static const char nm_label[] = "netmask";
```

Comment:
could you move these 3 variables on top of the file?

Reply:
sure